tasks/rabbitmq_vhosts.yml

```diff
@@ -1,8 +1,27 @@
 ---
-- name: Create vhosts
+- name: rabbitmq_extra_vhosts | Create vhosts
   rabbitmq_vhost:
     name: "{{ item['name'] }}"
     state: "{{ item['state'] }}"
   with_items: "{{ rabbitmq_extra_vhosts }}"
   run_once: "{{ rabbitmq_enable_clustering is defined and rabbitmq_enable_clustering }}"
   delegate_to: "{{ rabbitmq_master|default(omit) }}"
+  register: rabbitmq_created_vhosts
+
+- name: rabbitmq_extra_vhosts | Check guest administrator is present
+  command: rabbitmqctl -q list_users
+  become: true
+  run_once: "{{ rabbitmq_enable_clustering is defined and rabbitmq_enable_clustering }}"
+  delegate_to: "{{ rabbitmq_master|default(omit) }}"
+  when: rabbitmq_created_vhosts.changed
+  register: rabbitmq_existing_users
+
+- name: rabbitmq_extra_vhosts | Give access to new vhosts to guest administrator
+  command: "rabbitmqctl -q set_permissions -p {{ item['name'] }} guest '.*' '.*' '.*'"
+  become: true
+  run_once: "{{ rabbitmq_enable_clustering is defined and rabbitmq_enable_clustering }}"
+  delegate_to: "{{ rabbitmq_master|default(omit) }}"
+  with_items: "{{ rabbitmq_created_vhosts.results|selectattr('changed')|list }}"
+  when:
+    - item['state'] == 'present'
```

Comment:
Shouldn't there is a and with rabbitmq_created_vhosts.created ?

Reply:
let me check

Reply:
Maybe the with_items will result in an empty set

Reply:
Unfortunately there is no `created` flag. Cf:
```
    "rabbitmq_created_vhosts": {
        "changed": true,
        "msg": "All items completed",
        "results": [
            {
                "_ansible_delegated_vars": {
                    "ansible_delegated_host": "rabbit_01",
                    "ansible_host": "172.10.0.1"
                },
                "_ansible_ignore_errors": null,
                "_ansible_item_label": {
                    "name": "apps",
                    "state": "present"
                },
                "_ansible_item_result": true,
                "_ansible_no_log": false,
                "_ansible_parsed": true,
                "changed": false,
                "failed": false,
                "invocation": {
                    "module_args": {
                        "name": "apps",
                        "node": "rabbit",
                        "state": "present",
                        "tracing": false
                    }
                },
                "item": {
                    "name": "apps",
                    "state": "present"
                },
                "name": "apps",
                "state": "present"
            },
            {
                "_ansible_delegated_vars": {
                    "ansible_delegated_host": "rabbit_01",
                    "ansible_host": "172.10.0.1"
                },
                "_ansible_ignore_errors": null,
                "_ansible_item_label": {
                    "name": "paulrbr",
                    "state": "present"
                },
                "_ansible_item_result": true,
                "_ansible_no_log": false,
                "_ansible_parsed": true,
                "changed": true,
                "failed": false,
                "invocation": {
                    "module_args": {
                        "name": "paulrbr",
                        "node": "rabbit",
                        "state": "present",
                        "tracing": false
                    }
                },
                "item": {
                    "name": "paulrbr",
                    "state": "present"
                },
                "name": "paulrbr",
                "state": "present"
            }
        ]
    }
```

However checking `item['changed'] && item['state'] == 'present'` represents the `created` flag you expect :)